lib/elements/reinforcement/straight_longitudinal.rb

```diff
@@ -0,0 +1,80 @@
+require 'elements/reinforcement/base'
+require 'elements/reinforcement/straight_longitudinal_layer'
+require 'errors/reinforcement/empty_layers'
+
+module Elements
+  module Reinforcement
+    class StraightLongitudinal < Base
+      attr_accessor :layers
+
+      def initialize(z_base:, distribution_direction:, direction: 1)
+        @z_base = z_base
+        @direction = direction
+        @layers = []
+        @distribution_direction = distribution_direction
+      end
+
+      def modify_z_base(z_base:)
+        @z_base = z_base.to_f
+        update_z_base_of_layers unless @layers.empty?
+      end
+
+      def add_layer(start_location:, end_location:, number_of_rebars:, rebar:)
+        start_location.value_3 = @z_base
+        end_location.value_3 = @z_base
+        id = @layers.empty? ? 1 : @layers.last.id
+        @layers << Elements::Reinforcement::StraightLongitudinalLayer.new(
+          start_location:,
+          end_location:,
+          number_of_rebars:,
+          rebar:,
+          id:,
+          distribution_direction: @distribution_direction
+        )
+
+        @layers.last
+      end
+
+      def change(id_of_layer_to_change:, number_of_rebars:, rebar:)
+        straight_longitudinal_layer = find(id_of_layer_to_change)
+        straight_longitudinal_layer.modify(number_of_rebars:, rebar:)
+
+        straight_longitudinal_layer
+      end
+
+      def centroid_height
+        if @layers.empty?
+          raise Elements::Reinforcement::EmptyLayers, "can't complete centroid height calculation"
+        end
+
+        inertia = 0
+
+        @layers.each do |layer|
+          layer_centroid_height = @z_base + 0.5 * layer.diameter * @direction
```

Comment:
I think `@direction` should be an attribute of `StarightLongitudinalLayer`, and this class should expose a method to calculate its centroid height, it can receive the `@base_reference_height` as argument.

```ruby
layer_centroid_height = layer.centroid_height(@base_reference_height)
```

Reply:
can you take a look to [--> our wiki <-- Elements::Reinforcement::StraightLongitudinal #centroid_height section](https://www.notion.so/pradaing/Reinforcement-ff39ec7738fe4692a807ec9eaa1063ab#c36a4c5a76dd48b6920d809b8aa175a7)?